### ISSUE TITLE
failed to build 2.5.8 on macos sequoia

### ISSUE THREAD

**Issue by chenrui333** [falkTX/Carla#1926]
👋 trying to build 2.5.8 release on macos sequoia, but ran into the following build failure:

```
  Compiling juce_gui_basics.cpp
  In file included from juce_gui_basics.cpp:283:
  ./native/juce_mac_Windowing.mm:723:33: error: 'CGWindowListCreateImage' is unavailable: obsoleted in macOS 15.0 - Please use ScreenCaptureKit instead.
    723 |         CGImageRef screenShot = CGWindowListCreateImage (CGRectNull,
        |                                 ^
  /Library/Developer/CommandLineTools/SDKs/MacOSX15.sdk/System/Library/Frameworks/CoreGraphics.framework/Headers/CGWindow.h:271:33: note: 'CGWindowListCreateImage' has been explicitly marked unavailable here
    271 | CG_EXTERN CGImageRef __nullable CGWindowListCreateImage(CGRect screenBounds,
        |                                 ^
  Creating juce_core.a
  Compiling juce_gui_extra.cpp
  1 error generated.
```

full build log in here, https://github.com/Homebrew/homebrew-core/actions/runs/10823741563/job/30029771314#step:4:199

**Comment by chenrui333** [comment]
still seeing the same with 2.5.9 rel, https://github.com/Homebrew/homebrew-core/pull/191510 cc @falkTX

**Comment by falkTX** [comment]
I know, issue comes from juce which I have removed for the next major version but I dont want to have big changes in the 2.5.x so this is "wont fix" for 2.5.x

The next 2.6.x series is almost ready, but not quite yet

**Comment by chenrui333** [comment]
is there patch that I can backport it?

**Comment by falkTX** [comment]
> is there patch that I can backport it?

not really, I wrote new vst3 and AU hosting code so I wouldnt need to rely on juce for that.

and juce doesnt do LTS releases, the changes that handle sequoia are on JUCE8 which is quite the change for the old JUCE6 that 2.5.x Carla has

disabling JUCE is possible but that would drastically change how Carla works for a bugfix release (native AU and VST3 hosting dont work on 2.5.x and device handling is also JUCE)

I dont have a solution here besides building carla 2.5.x with older macOS.
For 2.6.x this wont be a problem, but will take a bit before that is ready.

**Comment by chenrui333** [comment]
got you, I will wait for the new 2.6.x release then. Thanks!
